xmppserver/src/main/java/org/jivesoftware/openfire/muc/spi/LocalMUCRoom.java

```diff
@@ -3283,7 +3283,7 @@ public void writeExternal(ObjectOutput out) throws IOException {
         ExternalizableUtil.getInstance().writeSafeUTF(out, description);
         ExternalizableUtil.getInstance().writeBoolean(out, canOccupantsChangeSubject);
         ExternalizableUtil.getInstance().writeInt(out, maxUsers);
-        ExternalizableUtil.getInstance().writeStringList(out, rolesToBroadcastPresence.stream().map(Enum::name).collect(Collectors.toList())); // This uses stringlist for compatibility with Openfire 4.6.0. Can be replaced the next major release.
```

Comment:
I'm confused. The comment said we're using StringList (List<String>) for backwards compatibility with 4.6.0, but why does xmppserver.jar v4.6.3 need to be compatible with xmppserver.jar 4.6.0? I thought this JAR only needs to be compatible with clients of its API f.e. plugins.

Reply:
The reasoning here is that this would not cause issues in a cluster where one cluster node has this update, but another does not (yet - eg when updating nodes online).